core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -4054,32 +4054,24 @@ struct RuntimeCheckViewMemoryAccessViolation<MemorySpace, AccessSpace, false> {
   template <class Track, class Map>
   KOKKOS_FUNCTION RuntimeCheckViewMemoryAccessViolation(char const* const msg,
                                                         Track const& track,
-                                                        Map const& map) {
-    char const* lbl;
-
-    char const unmanaged[]   = "**UNMANAGED**";
-    char const unavailable[] = "**UNAVAILABLE**";
+                                                        Map const&) {
+    char err[1024] = "";
```

Comment:
What was the UB?  `.c_str()` should be stable unless the source string is modified/destroyed.

Also, I prefer `strncat` over `strcat`, because if the source is too long, it gets truncated instead of UB.

Reply:
> What was the UB? .c_str() should be stable unless the source string is modified/destroyed.

Exactly

> Also, I prefer strncat over strcat, because if the source is too long, it gets truncated instead of UB.

Ok

Reply:
this is a bit iffy, you essentially increase the stack frame size for any bounds checking by 1kB, this may crash codes in Debug mode now which didn't before.

Reply:
They would need to increase the stack size for the GPU to make it work again, if they were close before.

Reply:
The stack frame size is only affected when there is illegal memory accesses in the code.
That said I agree 1 kilobyte is excessive.  I will reduce.